src/commonMain/kotlin/main.kt

```diff
@@ -568,3 +570,26 @@ private fun help(exeFile: File?, programName: String, supportedOptions: JaunchOp
     val optionsUnique = linkedSetOf(*supportedOptions.values.toTypedArray())
     optionsUnique.forEach { printlnErr(it.help()) }
 }
+
+private fun applyUpdate(appDir: File, updateSubDir: File) {
+    if (!updateSubDir.exists) return
+
+    // Recursively copy over all files in the update subdir
+    for (file in updateSubDir.ls()) {
+        if (file.isDirectory) applyUpdate(appDir, file)
+        else {
+            val dest = appDir / file.path.substring((appDir / "update").path.length)
+            if (file.stat() == 0L) {
```

Comment:
Why do we need to check `file.stat() == 0L` here? I feel like the `stat()` function is kinda too low level for this File interface. We also already have `isFile` and `isDirectory` and `exists`, which already use `stat` under the hood at least on POSIX.  Is the goal to check file existence? If so, can we use `file.exists` rather than `file.stat() == 0L`?

Reply:
This is the logic that has to check the file size.. if it's 0 then it signifies a deletion of the corresponding file, and if non-0 then it's a move. This was the mechanism ChatGPT gave me for checking file length... 😬

Reply:
Ah, I see! I'd prefer to add a `length` or `size` accessor to `File` rather than the more generic `stat`. Unless this "stat" function is really just file length? In which case we could simply rename it to `length`.